python/ray/train/huggingface/_huggingface_utils.py

```diff
@@ -84,7 +84,7 @@ def process_dataset_for_hf(dataset: Dataset) -> "IterableDataset":
     ).with_format("torch")
 
     try:
-        dataset_length = dataset.count()
+        dataset_length = dataset._base_dataset_or_pipeline.count()
```

Comment:
With datasets streaming by default, `count()` will trigger full execution of the dataset.

Is length required here, or can we go with the pipelined case as the default?

Reply:
Having the length would make for best UX. How about we discuss that in a followup?

Reply:
ok can we not use private attributes from DatasetIterator?

Reply:
How can I obtain the count in that case?

Reply:
ok that's fine, but let's make sure to follow up

Reply:
I'll make an issue